### ISSUE TITLE
Feature Request: Include sticky posts

### ISSUE THREAD

**Issue by dbrunsden** [dcooney/wordpress-ajax-load-more#25]
Please can you include the ability to make sticky posts work.

Thanks

**Comment by emathiasen** [comment]
Any update on this? This would be a great feature to have.

**Comment by webhanspeter** [comment]
Being able to filter on sticky posts is a must for us and we need it to be included before we can invest in this module.

**Comment by tizzle** [comment]
Hey, is there any update on this? This feature is very much needed for me.

Thanks

**Comment by dcooney** [comment]
This is on the feature list but not sure when I will get to it.

In the meantime, a solution might be to create a custom query before the `[ajax_load_more]` shortcode and pass array of id to the post__in param.

The following is untested.
```
<?php
$ids = get_option( 'sticky_posts' );
rsort( $ids );
/* Query sticky posts */
$the_query = new WP_Query( 
   array( 
      'post__in' => $sticky, 
      'ignore_sticky_posts' => 1,
      'posts_per_page' => -1,
   )
);
// The Loop
while ( $the_query->have_posts() ) {
	$the_query->the_post();
	$ids[] = get_id_ID();	
}
echo do_shortcode('[ajax_load_more post__in="'.$ids.'" orderby="post__In"]'); 
```

**Comment by ibanner** [comment]
@dcooney Couldn't get your code to work, so I changed it a bit and this one works for me:

```
$ids = get_option( 'sticky_posts' );
rsort( $ids );
/* Query sticky posts */
$the_query = new WP_Query(
    array(
        'post__not_in' => $ids,
        'posts_per_page' => -1,
    )
);
// The Loop
while ( $the_query->have_posts() ) {
    $the_query->the_post();
    $ids[] = get_the_ID();
}
$ids = implode(', ', $ids);
```

**Comment by Jungingen** [comment]
Yes, sticky posts would be very usefull. The last workaround works, but on page load it shows two loading images, instead of one. That's very anoying.

**Comment by dcooney** [comment]
@ibanner thanks. this looks good.

@Jungingen What do you mean?

**Comment by Jungingen** [comment]
@dcooney sorry, I got confused. I thought it was about using special shortcode for sticky posts and after that, the next one with the regular. My mistake.

**Comment by dcooney** [comment]
@Jungingen No worries.

**Comment by dcooney** [comment]
I've decided not to build sticky posts directly into the plugin. 

The example @ibanner provided works nice so I'm going to set up an example on the [Code Samples](https://connekthq.com/plugins/ajax-load-more/docs/code-samples/sticky-posts/) page using the [alm_query_args](https://connekthq.com/plugins/ajax-load-more/docs/filter-hooks/#alm_query_args) filter and his code.

**Comment by aaronlutze** [comment]
Thanks @ibanner, I used your code and modified it to make a custom shortcode in functions.php to get stickies working. Unfortunately @dcooney I wasn't able to get the code samples working on your site as per those links, particularly when wanting them in a form that is easy for my content creators to use without being devs with php coding knowledge.

I ended up creating my own derivative of everyones code above as a new function in functions.php and a new custom shortcode that allows for several of the alm properties to be set by a non-dev, and include support for stickies.

I think it would be a really big deal if stickies could be added in as a standard feature for ALM as this plugin is really great, and supporting a standard feature of WordPress would go a long way to make the plugin more user-friendly.
